react/State.js

```diff
@@ -0,0 +1,192 @@
+const gql = require('graphql-tag')
+
+const minicartItemsQuery = gql`
+  query {
+    minicart @client {
+      items {
+        upToDate
+        id
+        name
+        imageUrl
+        detailUrl
+        skuName
+        quantity
+        sellingPrice
+        listPrice
+        seller
+        index
+        parentItemIndex
+        parentAssemblyBinding
+      }
+    }
+  }
+`
+
+export const resolvers = {
+  Mutation: {
+    addToCart: (_, { items }, { cache }) => {
+      const query = minicartItemsQuery
+      const {
+        minicart: { items: prevItems },
+      } = cache.readQuery({ query })
+
+      const indexedItems = items.map(item => ({
+        index: prevItems.findIndex(({ id }) => id === item.id),
+        item: mapToMinicartItem({
+          ...item,
+          upToDate: false,
+        }),
+      }))
+
+      const newItems = []
+      for (const indexedItem of indexedItems) {
+        const { index, item } = indexedItem
+        if (index !== -1) {
+          prevItems[index] = item
+        } else {
+          newItems.push(item)
+        }
+      }
+
+      cache.writeData({
+        data: {
+          minicart: {
+            __typename: 'Minicart',
+            items: newItems.concat(prevItems),
+          },
+        },
+      })
+      return newItems.concat(prevItems)
+    },
+    updateItems: (_, { items: newItems }, { cache }) => {
+      const query = minicartItemsQuery
+      const {
+        minicart: { items: prevItems },
+      } = cache.readQuery({ query })
+
+      const items = prevItems
+        .map(prevItem => {
+          const newItem = newItems.find(({ id }) => id === prevItem.id)
+          return newItem ? { ...newItem, upToDate: false } : prevItem
+        })
+        .map(mapToMinicartItem)
+
+      cache.writeData({
+        data: {
+          minicart: { __typename: 'Minicart', items },
+        },
+      })
+      return items
+    },
+    updateOrderForm: (_, { orderForm: newOrderForm }, { cache }) => {
+      const orderForm = mapToLinkStateOrderForm(newOrderForm)
+      cache.writeData({
+        data: {
+          minicart: { __typename: 'Minicart', orderForm },
+        },
+      })
+      if (orderForm.items) {
+        updateLinkStateItems(orderForm.items, cache)
+      }
+      return orderForm
+    },
+  },
+}
+
+function updateLinkStateItems(newItems, cache) {
```

Comment:
Any reason to use `function` instead of other function's declarations with `const`?